### ISSUE TITLE
Build failure with feature simd-nightly

### ISSUE THREAD

**Issue by dhardy** [google/zerocopy#844]
```
$ rustc -V
rustc 1.78.0-nightly (256b6fb19 2024-02-06)
$ cargo check --features simd-nightly
warning: `/home/dhardy/.cargo/config` is deprecated in favor of `config.toml`
note: If you need to support cargo 1.38 or earlier, you can symlink `config` to `config.toml`
warning: `/home/dhardy/.cargo/config` is deprecated in favor of `config.toml`
note: If you need to support cargo 1.38 or earlier, you can symlink `config` to `config.toml`
    Checking zerocopy v0.8.0-alpha.3 (/home/dhardy/projects/rust/zerocopy)
error[E0658]: use of unstable library feature 'stdarch_x86_avx512'
    --> src/lib.rs:3719:43
     |
3719 |                   use core::arch::$arch::{$($typ),*};
     |                                             ^^^^
...
3746 | /         simd_arch_mod!(
3747 | |             #[cfg(all(feature = "simd-nightly", target_arch = "x86_64"))]
3748 | |             x86_64, x86_64_nightly, __m512bh, __m512, __m512d, __m512i
3749 | |         );
     | |_________- in this macro invocation
     |
     = note: see issue #111137 <https://github.com/rust-lang/rust/issues/111137> for more information
     = help: add `#![feature(stdarch_x86_avx512)]` to the crate attributes to enable
     = note: this compiler was built on 2024-02-06; consider upgrading it if it is out of date
     = note: this error originates in the macro `simd_arch_mod` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0658]: use of unstable library feature 'stdarch_x86_avx512'
    --> src/lib.rs:3722:38
     |
3722 |                   impl_known_layout!($($typ),*);
     |                                        ^
...
3746 | /         simd_arch_mod!(
3747 | |             #[cfg(all(feature = "simd-nightly", target_arch = "x86_64"))]
3748 | |             x86_64, x86_64_nightly, __m512bh, __m512, __m512d, __m512i
3749 | |         );
     | |_________- in this macro invocation
     |
     = note: see issue #111137 <https://github.com/rust-lang/rust/issues/111137> for more information
     = help: add `#![feature(stdarch_x86_avx512)]` to the crate attributes to enable
     = note: this compiler was built on 2024-02-06; consider upgrading it if it is out of date
     = note: this error originates in the macro `simd_arch_mod` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0658]: use of unstable library feature 'stdarch_x86_avx512'
    --> src/lib.rs:3726:37
     |
3726 |                       $( unsafe_impl!($typ: NoCell, TryFromBytes, FromZeros, FromBytes, IntoBytes); )*
     |                                       ^
...
3746 | /         simd_arch_mod!(
3747 | |             #[cfg(all(feature = "simd-nightly", target_arch = "x86_64"))]
3748 | |             x86_64, x86_64_nightly, __m512bh, __m512, __m512d, __m512i
3749 | |         );
     | |_________- in this macro invocation
     |
     = note: see issue #111137 <https://github.com/rust-lang/rust/issues/111137> for more information
     = help: add `#![feature(stdarch_x86_avx512)]` to the crate attributes to enable
     = note: this compiler was built on 2024-02-06; consider upgrading it if it is out of date
     = note: this error originates in the macro `simd_arch_mod` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0635]: unknown feature `stdsimd`
   --> src/lib.rs:223:47
    |
223 | #![cfg_attr(feature = "simd-nightly", feature(stdsimd))]
    |                                               ^^^^^^^

Some errors have detailed explanations: E0635, E0658.
For more information about an error, try `rustc --explain E0635`.
error: could not compile `zerocopy` (lib) due to 9 previous errors
```

**Comment by joshlf** [comment]
Thanks for letting us know! Should be fixed soon.

**Comment by joshlf** [comment]
The fix has been published as [0.8.0-alpha.4](https://crates.io/crates/zerocopy/0.8.0-alpha.4).
